step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java

```diff
@@ -331,7 +342,7 @@ static ParameterType of(Type type) {
                     }
                 }
                 if (acceptsList(type)) {
-                    return new ArrayType(of(((ParameterizedType) type).getActualTypeArguments()[0]));
+                    return new ArrayType(List.class, of(((ParameterizedType) type).getActualTypeArguments()[0]));
```

Comment:
Not sure what the difference would be between `Type` and `Class`... I'll switch to `Collection`, though, since that sounds right. More tweaking remains to be done on the parameter typing in the DSL.